runelite-mixins/src/main/java/net/runelite/mixins/RSDynamicObjectMixin.java

```diff
@@ -0,0 +1,54 @@
+package net.runelite.mixins;
+
+import net.runelite.api.mixins.Copy;
+import net.runelite.api.mixins.FieldHook;
+import net.runelite.api.mixins.Inject;
+import net.runelite.api.mixins.Mixin;
+import net.runelite.api.mixins.Replace;
+import net.runelite.api.mixins.Shadow;
+import net.runelite.rs.api.RSClient;
+import net.runelite.rs.api.RSDynamicObject;
+import net.runelite.rs.api.RSModel;
+
+@Mixin(RSDynamicObject.class)
+public abstract class RSDynamicObjectMixin implements RSDynamicObject
+{
+	@Shadow("clientInstance")
+	private static RSClient client;
+
+	@Copy("getModel")
+	public abstract RSModel rs$getModel();
+
+	@Replace("getModel")
+	public RSModel rl$getModel()
+	{
+		try
+		{
+			return rs$getModel();
+		}
+		finally
+		{
+			if (getAnimFrame() < 0)
+			{
+				setAnimFrame((getAnimFrame() ^ Integer.MIN_VALUE) & 0xFFFF);
+			}
+		}
+	}
+
+	@FieldHook("animCycleCount")
+	@Inject
+	public void onAnimCycleCountChanged(int idx)
+	{
+		boolean fromConstructor = true;
+		// hack to see if this got called from the constructor
+		StackTraceElement[] stackTrace = new Throwable().getStackTrace();
```

Comment:
Did you check how often this is run and if constructing this stracktrace will have an effect on performance/fps?

Reply:
I've added the license files and replaced the constructor check